src/test/java/com/amihaiemil/docker/RtNetworksTestCase.java

```diff
@@ -152,71 +150,57 @@ public void createWithParametersOk() throws Exception {
-                new Condition(
-                    "create() must send the request to the create url",
-                    req -> "http://localhost/networks/create".equals(
-                        req.getRequestLine().getUri()
+        Network network =
+            new ListedNetworks(
+                new AssertRequest(
+                    new Response(
+                        HttpStatus.SC_CREATED,
+                        Json.createObjectBuilder()
+                            .add("Id", "id1").build().toString()
+                    ),
+                    new Condition(
+                        "create() must send a POST HTTP request",
+                        req -> "POST".equals(req.getRequestLine().getMethod())
+                    ),
+                    new Condition(
+                        "create() must send the request to the create url",
+                        req -> "http://localhost/networks/create".equals(
+                            req.getRequestLine().getUri()
+                        )
+                    ),
+                    new Condition(
+                        "create() must send Json body request",
+                        req -> {
+                            final JsonObject payload = new PayloadOf(req);
+                            // @checkstyle LineLength (3 lines)
+                            return payload.getJsonObject("IPAM").getString("Driver").equals(ipam.getString("Driver"))
+                                && payload.getJsonObject("Options").getString("option1").equals(options.getString("option1"))
+                                && payload.getJsonObject("Labels").getString("label1").equals(labels.getString("label1"));
+                        }
                     )
-                )
-            ),
-            URI.create("http://localhost/networks"),
-            DOCKER
-        ).create(
-            "testwithparameters",
-            Json.createObjectBuilder()
-            .add("Driver", "networkdriver")
-            .add(
-                "IPAM",
-                ipam
-            ).add(
-                "Options",
-                options
-            ).add(
-                "Labels",
-                labels
-            ).build()
-        );
+                ),
+                URI.create("http://localhost/networks"),
+                DOCKER
+            ).create(
+                "testwithparameters",
+                Json.createObjectBuilder()
+                .add("Driver", "networkdriver")
+                .add(
+                    "IPAM",
+                    ipam
+                ).add(
+                    "Options",
+                    options
+                ).add(
+                    "Labels",
+                    labels
+                ).build()
+            );
         MatcherAssert.assertThat(
             "could not return correct network id",
             network.getString("Id"),
             new IsEqual<>("id1")
         );
-        MatcherAssert.assertThat(
```

Comment:
@bkuzmic I think that these assertions here are easier to understand than using that `Condition` construct

Reply:
@paulodamaso I agree, but here I'm testing the request payload and in the response I only get Id back.